runtime/src/main/cpp/ObjCExceptions.cpp

```diff
@@ -0,0 +1,125 @@
+#if KONAN_REPORT_BACKTRACE_TO_IOS_CRASH_LOG
+
+#include <dlfcn.h>
+#include <inttypes.h>
+#include <mach-o/loader.h>
+#include <CoreFoundation/CFRunLoop.h>
+#include "Natives.h"
+#include "ObjCExceptions.h"
+#include "Types.h"
+
+extern "C" OBJ_GETTER(Kotlin_Throwable_getStackTrace, KRef throwable);
+
+static void writeStackTraceToBuffer(KRef throwable, char* buffer, unsigned long bufferSize) {
+  if (bufferSize < 2) return;
+
+  ObjHolder stackTraceHolder;
+  ArrayHeader* stackTrace = Kotlin_Throwable_getStackTrace(throwable, stackTraceHolder.slot())->array();
+
+  char* bufferPointer = buffer;
+  unsigned long remainingBytes = bufferSize;
+
+  *(bufferPointer++) = '(';
+  --remainingBytes;
+
+  for (int index = 0; index < stackTrace->count_; ++index) {
+    KNativePtr ptr = *PrimitiveArrayAddressOfElementAt<KNativePtr>(stackTrace, index);
+    int bytes = snprintf(bufferPointer, remainingBytes, "0x%" PRIxPTR " ", reinterpret_cast<uintptr_t>(ptr));
+
+    if (bytes >= remainingBytes) {
+      break;
+    }
+
+    bufferPointer += bytes;
+    remainingBytes -= bytes;
+  }
+
+  *(bufferPointer - 1) = ')'; // Replace last space.
+  *bufferPointer = '\0';
+}
+
+#if !defined(MACHSIZE)
+#error "Define MACHSIZE to 32 or 64"
+#endif
+
+#if MACHSIZE == 32
+
+typedef struct mach_header mach_header_host;
```

Comment:
Technically, it's more like `_target` than `_host`.

Reply:
Maybe just 
```
#if MACHSIZE == 32
#define ARGH_SUFFIX(x) x
#else 
#define ARGH_SUFFIX(x) x##_64
#endif``

Reply:
I would prefer to avoid using macros this heavily.